Bottom: 0127efb585e3bcdf1d82d440b99ec714916e7cec
Top:    3701d3cee693d49621303abd7ba483eeacda363c
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-10-02 08:14:58 +0800

Refresh of introduce-core-ring

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 4fbd799..7728e13 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1185,6 +1185,7 @@ static int tun_xdp_xmit(struct net_device *dev, struct xdp_buff *xdp)
 		if (tfile->flags & TUN_FASYNC)
 			kill_fasync(&tfile->fasync, SIGIO, POLL_IN);
 		tfile->socket.sk->sk_data_ready(tfile->socket.sk);
+		return -ENOSPC;
 	}
 
 	return 0;
